### PR TITLE
Feature/structure refactoring

### PR DESCRIPTION
**Pull request by Aleksandra05sh**
Refactoring structure of the project (https://trello.com/c/Jb2IUNSp/6-refactoring-structure-of-the-project)

- moved colors to constants(mui theme). Import from scss to ts.
- added editorconfig and provide eol
- added Eslint-plugin-simple-import-sort to  import sorting
- added redux structure (using configureStore )

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,12 +24,13 @@
...
     "react": "^18.0.0",
     "react-dom": "^18.0.0",
+    "react-redux": "^8.0.1",
     "react-router": "^6.3.0",
     "react-router-dom": "^6.3.0",
     "react-scripts": "^5.0.0",
-    "redux": "^4.1.2",
     "sass": "^1.49.11",
     "styled-components": "^5.3.5",
+    "swiper": "^8.1.4",
```
Comment (vovkis13) [inline package.json:33]: we use swiper 6.0.2

Comment (vovkis13) [inline package.json:33]: we use swiper 6.0.2

File: src/shared/styles/_variables.scss [hunk 2]
```diff
@@ -2,6 +2,10 @@
 $primary-main: #90caf9;
 $primary-light: #c3fdff;
 $primary-dark: #5d99c6;
+$primary-contrastText: #212121;
+$secondary-main: #ffee58;
+$secondary-light: #ffff8b;
+$secondary-dark: #c9bc1f;
 $contrast-text-color: #212121;
 $timer-border-color: #21212180;
 $primary-main-color: #90caf9;
```
Comment (topsart) [inline src/shared/styles/_variables.scss:10]: I think we can join identical values

File: src/components/Header/Header.tsx [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import React from 'react';
-import { AppBar, Toolbar, Container } from '@mui/material';
+import { AppBar, Container, Toolbar } from '@mui/material';
 import { NavigationList } from './components/NavigationList';
-import { Logo } from './components/Logo';
 import { SocialsList } from './components/SocialsList';
+import { Logo } from './components/Logo';
 
 export const Header = () => {
   return (
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply src/components/Header/Header.tsx:8]: Thank you 
it was the previous PR which already merged
I will fix it in the next story

File: src/store/template/reducer.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import { TemplateActionTypes } from './actionTypes';
+import type { TemplateState } from '../types';
+
+const initialState: TemplateState = {
+  name: '',
+  value: 0,
+};
+
+export const templateReducer = (state = initialState, action: AnyAction) => {
+  const { type, payload } = action;
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply src/store/template/reducer.ts:11]: I prefer this option, but we can  discuss it with our team to make it consistent for the whole project

File: src/store/app/reducer.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import type { AppState } from '../types';
+
+//TODO: update with real values
+const initialState: AppState = {
+  name: '',
+  value: 0,
+};
+
+//for example, will be changed
+export const appReducer = (state = initialState, action: AnyAction) => {
```
Comment (kerrimov) [inline src/store/app/reducer.ts:12]: AnyAction can’t be used. Use your own interface for it

**Comment by kerrimov** [comment]
`src/store/template` folder should be deleted from the PR

File: README.md [hunk 6]
```diff
@@ -21,3 +22,39 @@ _Launches the test runner in the interactive watch mode._
 _Builds the app for production to the build folder._
 
 ---
+
+## Code convention
+
+### Syntax
+
+- Keep code lines under 80 columns wide..
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply README.md:30]: corrected

File: README.md [hunk 7]
```diff
@@ -21,3 +22,39 @@ _Launches the test runner in the interactive watch mode._
...
+- Use soft tabs set to 2 spaces.
+- Indent lines with tabs instead of spaces.
+- Use single quotes.
+- Use trailing commas wherever possible, including function parameters and
+  calls.
+- Print spaces between brackets in object literals.
+- Put the > of a multi-line JSX element on the next line.
+- Exclude parentheses around a sole arrow function parameter.
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply README.md:38]: this info was taken from .prettierrc.json:
1. "bracketSpacing": true  = Print spaces between brackets in object literals.  https://prettier.io/docs/en/options.html#bracket-spacing

2.   "jsxBracketSameLine": false  = Put the > of a multi-line JSX element on the next line.  
https://prettier.io/docs/en/options.html#deprecated-jsx-brackets

3. "arrowParens": "avoid" = Exclude parentheses around a sole arrow function parameter. 
https://prettier.io/docs/en/options.html#arrow-function-parentheses

File: README.md [hunk 8]
```diff
@@ -21,3 +22,39 @@ _Launches the test runner in the interactive watch mode._
...
+  etc.). Place no space before the argument list in function calls and
+  declarations.
+- Set off operators with spaces.
+
+### Style
+
+- Use Sass to keep large stylesheets well-organized.
+- Favor rems for relative values, preferably typography.
+- Don’t be afraid of using px when you want to exactly define a measure.
+- Don't abuse !important.
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply README.md:53]: corrected

File: src/store/template/reducer.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (kerrimov) [reply src/store/template/reducer.ts:11]: Both options are ok

File: src/components/Router/Router.tsx [hunk 9]
```diff
@@ -1,9 +1,9 @@
 import React from 'react';
-import './Router.scss';
 import { useRoutes } from 'react-router-dom';
 import { routesConfig } from './routes';
+import './Router.scss';
 
 export const Router = () => {
-  const routes: React.ReactElement | null = useRoutes(routesConfig);
+  const routes: React.ReactElement = useRoutes(routesConfig) || <></>;
```
Comment (kerrimov) [inline src/components/Router/Router.tsx:7]: you can use Generic in the hook. Something like this `useRoute<YourOwnType>`

File: src/store/app/reducer.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
+// import { AnyAction } from 'redux';
```
Comment (kerrimov) [inline src/store/app/reducer.ts:1]: remove import

File: src/store/app/reducer.ts [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+//TODO: update with real values
+const initialState: AppState = {
+  name: '',
+  value: 0,
+};
+interface AppAction {
+  type: AppActionTypes;
+  payload?: AppState;
+}
```
Comment (kerrimov) [inline src/store/app/reducer.ts:13]: this interface should be moved to types

File: src/store/app/reducer.ts (hunk 10 above)
Comment (Aleksandra05sh) [reply src/store/app/reducer.ts:1]: sorry,  done

File: src/components/Router/Router.tsx (hunk 9 above)
Comment (Aleksandra05sh) [reply src/components/Router/Router.tsx:7]: using generic is not provided by this hook

image

File: src/store/app/reducer.ts (hunk 11 above)
Comment (Aleksandra05sh) [reply src/store/app/reducer.ts:13]: sorry, done

File: src/components/Router/Router.tsx (hunk 9 above)
Comment (kerrimov) [reply src/components/Router/Router.tsx:7]: Please use this doc https://reactnavigation.org/docs/typescript/#annotating-useroute

Comment (Aleksandra05sh) [reply src/components/Router/Router.tsx:7]: we are using react router, not react navigation 

https://reactrouter.com/docs/en/v6/api#useroutes

Comment (kerrimov) [reply src/components/Router/Router.tsx:7]: This is just an example how to use typescript

File: src/store/app/reducer.ts (hunk 10 above)
Comment (kerrimov) [reply src/store/app/reducer.ts:1]: don't see changes

File: src/components/Router/Router.tsx (hunk 9 above)
Comment (kerrimov) [reply src/components/Router/Router.tsx:7]: The example from the react-router: https://reactrouter.com/docs/en/v6/api#routes-and-route

Comment (kerrimov) [reply src/components/Router/Router.tsx]: Leave the approach that you had used before.

**Review by kerrimov — APPROVED** [review]

**Pull request merged**